packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_datetime.dart

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'temporal.dart';
+
+/// Represents a valid extended ISO-8601 DateTime string.  The time zone offset is compulsory.
+/// YYYY-MM-DDThh:mm:ss.sssZ  (ISO_OFFSET_DATE_TIME)
+/// https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html#appsync-defined-scalars
+class TemporalDateTime {
```

Comment:
If these types can be used across categories (say, in API.Graphql) then perhaps they should live in amplify_core? What do others think?

Reply:
Ideally, `amplify_core` should be kept small, since it is a dependency of all the plugins.   Moving the `Temporal` types and other API concepts to `amplify_core` will likely result in bloat over time.  In `amplify-android`, `Temporal.java` lives in a module called `aws-api-appsync`, which is a module containing all the common code needed by `aws-api` and `aws-datastore`.  Could a similar pattern work for Flutter?

Reply:
Moving to a separate `aws-api-appsync` module makes sense to me.  This will change our imports so it will change the codegen models as well.  
@haverchuck @Amplifiyer @Ashish-Nanda what are your thoughts on making that change?

Reply:
Can't we just re-export it from here? If yes, We can hold off on doing the refactor until API category needs this and move it then to what makes most sense at that time. If it is going to be a breaking change, then let's figure it out right now.

Reply:
@Amplifiyer 

I believe it will be a breaking change because the dart file imports will change for the codegen models: 

import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';

will change to whatever new package it's in

Reply:
If the new package gets exported from `amplify_datastore_plugin_interface.dart` then nothing will need to change, right?